src/Map.php

```diff
@@ -241,7 +241,12 @@ private function getHash($key)
             // Take a hash of the serialized value to avoid PHP having to do a
             // long string comparison on each lookup (we already check the
             // actual array values for equality ourselves).
-            return md5(serialize($key));
+            try {
+                return md5(serialize($key));
+            } catch (\Exception $e) {
+                // Sometimes serialization fails because the array contains anonymous classes or closures.
+                return md5(serialize(array_map(fn($val) => $this->getHash($val), $key)));
```

Comment:
Wow. I had no idea it was doing this when you use an array as a key. Ugh. Maybe we can go back to using the native extension?

Reply:
https://github.com/php-ds/ext-ds/blob/d42750d69beb684ec3ef7b3da48fba769ce57ffc/src/ds/ds_htable.c#L219-L239

Reply:
Oh. :-(

The message is: don't use large arrays as keys.